### ISSUE TITLE
New UI for finding/creating rooms (including bridged rooms)

### ISSUE THREAD

**Issue by ara4n** [vector-im/riot-web#2054]
This is split off from https://github.com/vector-im/vector-web/issues/1882.

@antikewl has provided some designs for finding/creating rooms as per zpl.io/11c9Ws, zpl.io/Z1u6JH8, https://zpl.io/2jFojg and https://zpl.io/2erp6m.

There are some problems here though:
- This makes the room directory incredibly visible when you want to create a room.  Psychologically, if your intent is to create a private room (e.g. Vector Design), and you hit the Rooms button, you're currently shown all the directory contents by default - immediately making you think in terms of public free-for-all rooms.  The directory also takes a while to load.  I am unsure that we want to be showing it by default, but only if the user explicitly says "actually, i want to find a public room to chat in".
- We need a way of filtering the directory to particular remote networks - e.g. "show me available channels on the ribot slack".  In future, we will also need the ability to select different directories - e.g. "show me rooms from the arasphere.net Matrix server" or "show me rooms for the Vector team group"
- We need a way to join rooms by alias (currently this is cryptically hidden - if you enter an alias like #matrix:matrix.org into RoomDirectory's input field and hit return, it joins you).  Sorry for not spelling this out as a requirement earlier.

I don't have a concrete suggestion here, i'm afraid, but perhaps a page of form...

```
* [ Create a new room ]
* Join room: [ e.g. #foo:matrix.org........... ]   (
* Search room: [ .............. ]  [ List of directories |v|]  [ Directory ]

[ Search results or directory contents go here ]
```

...might be sufficient (square brackets indicate form controls), where hitting the Directory would show you the unfiltered directory contents, whereas Search would let you search the directory?

**Comment by antikewl** [comment]
![chat group 1](https://cloud.githubusercontent.com/assets/3658981/18272289/ec8f1f20-742e-11e6-8b47-09b255ccb524.png)
https://zpl.io/11c9Ws

Rooms default view
![chat directory search](https://cloud.githubusercontent.com/assets/3658981/18272301/fc0d8fc2-742e-11e6-800f-69e9c25b8e81.png)
https://zpl.io/ZCjdQJ

Rooms search result
![chat directory search result](https://cloud.githubusercontent.com/assets/3658981/18272319/1b1d7486-742f-11e6-96f0-41c4b89207f7.png)
https://zpl.io/ZjrXQV

Rooms network select/filter menu
![chat directory search result network select](https://cloud.githubusercontent.com/assets/3658981/18272339/36fcf7f8-742f-11e6-9b4c-a33991a5cf6b.png)
https://zpl.io/Z1RCiSd

Left menu search ("More rooms" opens pre-filled search result above)
![04_06 search room filter](https://cloud.githubusercontent.com/assets/3658981/18272363/5d91d488-742f-11e6-98eb-49f78bd6a33f.png)
https://zpl.io/Z13pkVG

**Comment by antikewl** [comment]
Revised left navigation
![chat left panel a](https://cloud.githubusercontent.com/assets/3658981/18467700/eb51c9e0-7998-11e6-8b03-300f066f0b6d.png)

Future room directory screen
![chat directory search b](https://cloud.githubusercontent.com/assets/3658981/18467717/0284e4d0-7999-11e6-8a0a-e0b9908251a4.png)

**Comment by ara4n** [comment]
A first pass of this is being implemented in #2178.  I'm going to keep this open at lower priority as we will probably need to iterate a bit more post-freeze.

**Comment by turt2live** [comment]
We've since iterated on the designs for post-1.0. The designs look awesome for Vector, though they are a bit dated now :(
